addons/html_editor/static/src/editor/core/delete_plugin.js

```diff
@@ -285,40 +301,51 @@ export class DeletePlugin extends Plugin {
 
         // Remove nodes
         let allNodesRemoved = true;
-        let nRemovedNodes = 0;
+        let anyNodesRemoved = false;
         for (const node of nodesToRemove) {
             const didRemove = this.removeNode(node);
             allNodesRemoved &&= didRemove;
-            nRemovedNodes += Number(didRemove);
+            anyNodesRemoved ||= didRemove;
         }
-        return { startRemoveIndex, allNodesRemoved, nRemovedNodes };
+        return { startRemoveIndex, allNodesRemoved, anyNodesRemoved };
     }
 
-    // Returns true if node was removed, false otherwise.
-    removeNode(node) {
-        // @todo @phoenix: get list of callbacks from resources
-        const handleUnremovable = [
-            {
-                callback: (node) => {
-                    // @todo @phoenix: break unremovable into the concerned plugins
-                    // @todo @phoenix: handle contains unremovable
-                    if (isUnremovable(node)) {
-                        node.replaceChildren();
-                        // @todo: not sure about this.
-                        fillEmpty(node);
-                        return true;
-                    }
-                },
-            },
-        ];
+    // The root argument is used by some predicates in which a node is
+    // conditionally unremovable (e.g. a table cell is only removable if its
+    // ancestor table is also being removed).
+    isUnremovable(node, root = undefined) {
+        // For now, there's no use case of unremovable text nodes.
+        // Should this change, the predicates must be adapted to take a Node
+        // instead of an Element as argument.
+        if (node.nodeType === Node.TEXT_NODE) {
+            return false;
+        }
+        if (node.nodeType !== Node.ELEMENT_NODE) {
+            return true;
+        }
+        return this.resources.unremovables.some((predicate) => predicate(node, root));
+    }
 
-        for (const { callback } of handleUnremovable) {
-            if (callback(node)) {
+    // Returns true if the entire subtree rooted at node was removed.
+    // Unremovable nodes take the place of removable ancestors.
+    removeNode(node) {
+        const root = node;
+        const remove = (node) => {
+            for (const child of [...node.childNodes]) {
```

Comment:
childNodes is iterable
```suggestion
            for (const child of node.childNodes) {
```

Reply:
`node.childNodes` returns a live NodeList. I need it frozen, as I'm making changes to its children.